--- src/Makefile.in	2020-04-23 22:40:39.000000000 +0200
+++ src/Makefile.in	2021-02-06 17:20:42.256590466 +0100
@@ -60,7 +60,13 @@
-all: $(OUTFILES)
+all: buildinf depend app $(OUTFILES)

-netmud: $(O_FILES)
+app: netmud ssl_slave info_slave portmsg
+
+netmud: $(O_FILES)
 	@echo "Making netmud."
  
-	-mv -f netmud netmud~
+
+ @if [ -f ${GAMEDIR}/netmud ]; then \
+  	-mv -f netmud netmud~
+		fi
+
-	$(CC) $(CCFLAGS) -o netmud $(O_FILES) $(LDFLAGS) $(SQL_LDFLAGS) $(LIBS)
+	$(CC) $(CCFLAGS) -o netmud $(O_FILES) $(LDFLAGS) $(SQL_LDFLAGS) $(LIBS)

@@ -83,6 +89,10 @@

 buildinf:
-	-rm -f ../hdrs/buildinf.h
+
+	@if [ -f ../hdrs/buildinf.h ]; then \
+	  -rm -f ../hdrs/buildinf.h
+	fi
+

@@ -88,1 +98,3 @@
+
+mksrc: funlocal flaglocal cmdlocal local patches potfile bflags htmltab lmathtab jsontypes
+
